packages/ra-test/src/TestContext.stories.tsx

```diff
@@ -0,0 +1,33 @@
+import * as React from 'react';
+import { useGetOne, DataProviderContext } from 'ra-core';
+
+import TestContext from './TestContext';
+
+export default { title: 'ra-test/TestContext' };
+
+const dataProvider = {
+    getOne: () =>
+        Promise.resolve({
+            data: {
+                id: 1,
+                title: 'foo',
+            },
+        }),
+} as any;
+
+const Book = ({ id }) => {
+    const { data, loaded } = useGetOne('books', id);
+    return loaded ? <span>{data.title}</span> : <span>loading</span>;
+};
+
+export const WithDataProvider = () => (
+    <TestContext
+        enableReducers={true}
+        // FIXME: Resources must be initialized to allow dataProvider hooks to work
```

Comment:
That's the heart of the problem. To be fixed by moving away from Redux for query cache